### ISSUE TITLE
[Bug]: InfoLabel's text should trap focus by default

### ISSUE THREAD

**Issue by adamsamec** [microsoft/fluentui#28199]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
N/A
```

### Are you reporting Accessibility issue?

yes

### Reproduction

N/A

### Bug Description

For screen reader users not to be confused what content belongs to the info text displayed when info button is activated, we should trap keyboard focus for the InfoButton's text by default. This is even more important in cases where aria-owns is not set for the InfoButton parent element inserting the info text right after the InfoButton within the accessibility tree, because in case the info text is inserted  at the end of the accessibility tree, the keybaord user might not discover it easily, or will be unable to go back to the trigger button.

Note that if focus is trapped, it is also advisable to include a Close/Cancel button to provide an easy and discovarable way for users to dismiss the popover.

### Requested priority

Blocking

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.s
